lib/cadet/chat/room.ex

```diff
@@ -0,0 +1,89 @@
+defmodule Cadet.Chat.Room do
+  @moduledoc """
+  Contains logic pertaining to chatroom creation to supplement ChatKit, an external service engaged for Source Academy.
+  ChatKit's API can be found here: https://pusher.com/docs/chatkit
+  """
+
+  require Logger
+
+  import Cadet.Chat.Token
+  import Ecto.Query
+
+  alias Cadet.Repo
+  alias Cadet.Assessments.{Answer, Submission}
+  alias Cadet.Accounts.User
+
+  @instance_id :cadet |> Application.fetch_env!(:chat) |> Keyword.get(:instance_id)
+
+  @doc """
+  Creates a chatroom for every answer, and updates db with the chatroom id.
+  Takes in Submission struct
+  """
+  def create_rooms(%Submission{
+        id: id,
+        student_id: student_id,
+        assessment_id: assessment_id
+      }) do
+    student = User |> where(id: ^student_id) |> Repo.one()
+
+    Answer
+    |> where(submission_id: ^id)
+    |> Repo.all()
+    |> Enum.filter(fn answer -> answer.comment == "" or answer.comment == nil end)
+    |> Enum.each(fn answer ->
+      case create_room(assessment_id, answer.question_id, student) do
```

Comment:
Simplify

```elixir
 |> Enum.each(fn answer ->
  with {:ok, %{"id" => room_id}} <- create_room(assessment_id, answer.question_id, student) do
    # something
  end
end)

Reply:
Moved the room creation to `answer_question/3`. I think it makes more sense to have it after an answer is inserted or updated successfully.